test/e2e-tests/support/setup.go

```diff
@@ -99,6 +132,37 @@ func RunSetup(ctx context.Context, config *Config) error {
 		if err != nil {
 			return err
 		}
+
+		if config.GitFetchConfig.SelfSignedCert != "" {
+			templateBytes, err := os.ReadFile(SelfSignedCertTemplatePath)
+			if err != nil {
+				return err
+			}
+
+			certBytes, err := os.ReadFile(config.GitFetchConfig.SelfSignedCert)
+			if err != nil {
+				return err
+			}
+
+			configMap := applyconfigv1.ConfigMapApplyConfiguration{}
+			err = yaml.Unmarshal(templateBytes, &configMap)
+			if err != nil {
+				return err
+			}
+
+			// name is set here because the configmap template is
+			// used for both the git_fetch ssc and the s3 one
+			configMapName := "git-self-signed-cert"
+			configMap.Name = &configMapName
+			configMap.Data = map[string]string{
+				"ca-bundle.crt": string(certBytes),
+			}
+
+			_, err = config.Clients.Kubernetes.CoreV1().ConfigMaps(config.Namespace).Apply(ctx, &configMap, metav1.ApplyOptions{FieldManager: "Apply"})
+			if err != nil {
+				return err
+			}
```

Comment:
IMHO this is unnecessary duplication and can be extracted to a separate function.